db.go

```diff
@@ -129,6 +131,53 @@ func SmartUnmarshal[I any](respond interface{}, wrapperError error) (data I, err
 	return data, err
 }
 
+// Used for define table name, it has no value.
+type Basemodel struct{}
+
+// SmartUnmarshal can be used with all DB methods with generics and type safety.
+// Will handle errors, can use table struct tag with `BaseModel` type.
+// What Try to do if that fail what will happen.
+// 1. If there are some ID on struct it will file table with ID
+// 2. If there are struct tags with type of `Basemodel` it will use values of that`
+// 3. If everything failed or not exist it will use struct name as tablename.
+func SmartMarshal[I any](inputfunc interface{}, data I) (output interface{}, err error) {
```

Comment:
Change
```
func SmartMarshal[I any](inputfunc interface{}, data I) (output interface{}, err error) {
```
To
```
func SmartMarshal[I any](inputfunc func(string, interface{}) (interface{}, error), data I) (output interface{}, err error) {
```

Reply:
giving error when replacing the line when the test case is `db.Select` or `db.Delete` Both of them have one parameter as input.


`cannot use db.Select (value of type func(what string) (interface{}, error)) as func(string, interface{}) (interface{}, error) value in argument to surrealdb.SmartMarshal
`